docs/architecture/tls13-experimental.md

```diff
@@ -0,0 +1,36 @@
+Experimental TLS 1.3 develpoments
+=================================
+
+Overview
+--------
+
+Mbed TLS doesn't support the TLS 1.3 protocol yet, but a prototype is in development.
+Stable parts of this prototype that can be independently tested are being successively
+upstreamed under the guard of the following macro:
+
+```
+MBEDTLS_SSL_PROTO_TLS1_3_EXPERIMENTAL
+```
+
+This macro will likely be renamed to `MBEDTLS_SSL_PROTO_TLS1_3` once a minimal viable
+implementation of the TLS 1.3 protocol is available.
+
+See the [documentation of `MBEDTLS_SSL_PROTO_TLS1_3_EXPERIMENTAL`](../../include/mbedtls/config.h)
+for more information.
+
+Status
+------
+
+The following lists which parts of the TLS 1.3 prototype have already been upstreamed
```

Comment:
Hm, good question. It's a bit different in that it's not guarded by `MBEDTLS_SSL_PROTO_TLS1_3_EXPERIMENTAL` because it makes sense to be used on its own. Still it should be mentioned. I will add something.